src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1560,18 +1573,105 @@ void AdaptDispatch::_InitTabStopsForWidth(const size_t width)
 }
 
 //Routine Description:
-// Designate Charset - Sets the active charset to be the one mapped to wch.
-//     See DispatchTypes::VTCharacterSets for a list of supported charsets.
-//     Also http://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-Controls-beginning-with-ESC
-//       For a list of all charsets and their codes.
+// DOCS - Selects the coding system through which character sets are activated.
+//     When ISO2022 is selected, the code page is set to ISO-8859-1, and both
+//     GL and GR areas of the code table can be remapped. When UTF8 is selected,
+//     the code page is set to UTF-8, and only the GL area can be remapped.
+//Arguments:
+// - codingSystem - The coding system that will be selected.
+// Return value:
+// True if handled successfully. False otherwise.
+bool AdaptDispatch::DesignateCodingSystem(const wchar_t codingSystem)
+{
+    // If we haven't previously saved the initial code page, do so now.
+    // This will be used to restore the code page in response to a reset.
+    if (_initialCodePage == 0)
+    {
+        _pConApi->GetConsoleOutputCP(_initialCodePage);
+    }
+
+    bool success = false;
+    switch (codingSystem)
+    {
+    case DispatchTypes::CodingSystem::ISO2022:
+        success = _pConApi->SetConsoleOutputCP(28591);
```

Comment:
wow, is there a constant for this?

Reply:
I'd love for there to be one, but I couldn't find anything. I think a lot of the code pages are just named after their number, so technically this is just "Windows-28591". In ISO terms it's ISO-8859-1, which isn't a whole lot more meaningful. I've said in the doc comment that that's the ISO code page we're selecting here. I'm not sure what else I could add.